modules/navier_stokes/src/actions/INSAction.C

```diff
@@ -173,7 +176,11 @@ INSAction::INSAction(InputParameters parameters)
     _fe_type(Utility::string_to_enum<Order>(getParam<MooseEnum>("order")),
              Utility::string_to_enum<FEFamily>(getParam<MooseEnum>("family"))),
     _use_ad(getParam<bool>("use_ad")),
-    _temperature_variable_name(getParam<VariableName>("temperature_variable"))
+    _temperature_variable_name(getParam<VariableName>("temperature_variable")),
```

Comment:
Will user provided temperature variable name consistent with the name internally used by kernels?

Reply:
Which kernels? The INS kernels? We are setting the `variable` parameter equal to the value of whatever the user sets here for `_temperature_variable_name`. You wrote this code

Reply:
I mean users set `add_temperature_variable` to false but did not provide `temperature_variable` which will cause the action to use the default temperature variable name, which could be inconsistent with the temperature name used by users. Am I missunderstanding something?

Reply:
Also moose can check if a variable has existed or not. `add_navier_stokes_variables` happens after `add_variables` if I remember correctly, so we can simply check if variable exists or not. If users want to have their own variable added through moose syntax, then can provide the variable name through `temperature_variable`, the action can do the right thing. We do not need an extra parameter for this.

Reply:
I changed it to work via your suggestion :+1: